src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/AutoInsert/AutoInsertService.cs

```diff
@@ -0,0 +1,65 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT license. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Frozen;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.AspNetCore.Razor.Language;
+using Microsoft.AspNetCore.Razor.PooledObjects;
+using Microsoft.VisualStudio.LanguageServer.Protocol;
+
+namespace Microsoft.CodeAnalysis.Razor.AutoInsert;
+
+internal class AutoInsertService(IEnumerable<IOnAutoInsertProvider> onAutoInsertProviders) : IAutoInsertService
```

Comment:
Sure, I'll go ahead and do that.

In general, I was trying to move the code as is without making too many changes unless necessary so we can get to parity in cohosting with what we currently have. I'd like to minimize the general cleanup changes at this point, especially since this PR has been around for over a month already, and address any general cleanup changes in a separate PR if needed so this one can be finished in a finite amount of time :)

Reply:
Actually, after giving it a try, it causes more changes than I would like to add to this PR, especially in tests. If we want to address this, I'd rather address it in a separate PR. It's certainly doable. Unless there is strong pushback, I won't be changing passing in providers via DI in this PR.